catalog/plugin/catalog-plugin-metadatabackup/src/main/java/org/codice/ddf/catalog/plugin/metadata/backup/MetadataBackupPlugin.java

```diff
@@ -0,0 +1,313 @@
+                copyMetadataToOutputDirectory(binaryContent, metacard.getId());
+            } catch (CatalogTransformerException e) {
+                LOGGER.debug("Unable to transform metacard with id {}.", metacard.getId(), e);
+            }
+        });
+    }
+
+    private void copyMetadataToOutputDirectory(BinaryContent content, String metacardId) {
+        if (content == null || content.getInputStream() == null) {
+            LOGGER.debug("No content for transformed metacard {}.", metacardId);
+            return;
+        }
+
+        File parent = getCompleteDirectory(getFolderDepth(), metacardId);
+        if (!parent.exists()) {
+            try {
+                FileUtils.forceMkdir(parent);
+            } catch (IOException e) {
+                LOGGER.debug("Unable to make directory {}", parent.getPath(), e);
+            }
+        }
+
+        validateDirectory(parent);
+        File file = new File(parent.getPath() + File.separator + metacardId);
+        LOGGER.trace("Writing metadata from {} to file {}", metacardId, file.getPath());
+
+        try (OutputStream outputStream = new FileOutputStream(file)) {
+            IOUtils.write(content.getByteArray(), outputStream);
+        } catch (IOException e) {
+            LOGGER.warn("Unable to backup {} to {}.  The directory may be full.",
+                    metacardId,
+                    file.getAbsolutePath(),
+                    e);
+        }
+    }
+
+    private void deleteMetadataIfPresent(String filename) {
+        File parent = getCompleteDirectory(getFolderDepth(), filename);
+        validateDirectory(parent);
+        File file = new File(parent.getPath() + File.separator + filename);
+        LOGGER.trace("Deleting metadata file {}", file.getPath());
+        try {
+            Files.deleteIfExists(file.toPath());
+        } catch (IOException e) {
+            LOGGER.debug("Unable to backup metadata {}. ", filename, e);
+        }
+        clearEmptyDirectories(parent);
+    }
+
+    private File getCompleteDirectory(int depth, String id) {
+        File parent = new File(getOutputDirectory());
+        for (int i = 0; i < depth && id.length() > (i * 2 + 2); i++) {
+            parent = new File(parent, id.substring(i * 2, i * 2 + 2));
+        }
+        return parent;
+    }
+
+    private void clearEmptyDirectories(File file) {
+        while (file != null && !file.getPath()
+                .equals(outputDirectory)) {
+            if (isDirectoryEmpty(file.toPath())) {
+                try {
+                    FileUtils.deleteDirectory(file);
+                } catch (IOException e) {
+                    LOGGER.debug("Unable to delete directory {}", file.toPath(), e);
+                }
+            }
+            file = file.getParentFile();
+        }
+    }
+
+    private boolean isDirectoryEmpty(Path dir) {
+        try (DirectoryStream<Path> dirStream = Files.newDirectoryStream(dir)) {
```

Comment:
We're mixing the "Path" way of doing things with the old "File" way of doing things. Like my previous comment says, you should probably switch everything over to using Paths (which this code is doing).